invisible_cities/reco/icaro_components.py

```diff
@@ -0,0 +1,219 @@
+                             input_mask   : Optional[np.ndarray]  = None                 ,
+                             eff_interval : Optional[Tuple[float, float]] = [0,1]        ,
+                             strictness   : Optional[Strictness] = Strictness.raise_error
+                             )->np.ndarray:
+    """
+    Selects nS1(or nS2) == 1 for a given kr dst and
+    returns the mask. It also computes selection efficiency,
+    checking if the value is within a given interval.
+    Parameters
+    ----------
+    dst: pd.Dataframe
+        Krypton dst dataframe.
+    column: type_of_signal
+        The function can be appplied over nS1 or nS2.
+    input_mask: np.array (Optional)
+        Selection mask of the previous cut. If this is the first selection
+        /no previous maks is input, input_mask is set to be an all True array.
+        Defaults to None.
+    interval: length-2 tuple (Optional)
+        If the selection efficiency is out of this interval
+        the map production will abort/just warn, depending on "strictness".
+        Defaults to [0, 1].
+    strictness: Strictness (Optional)
+        If 'warning', function returns a False if the criteria
+        is not matched. If 'stop_proccess' it raises an exception.
+        Defaults to `raise_error`.
+    Returns
+    ----------
+        A mask corresponding to the selected events.
+    """
+    input_mask = input_mask if input_mask is not None else [True] * len(dst)
+    mask             = np.zeros_like(input_mask)
+    mask[input_mask] = dst.loc[input_mask, column.value] == 1
+
+    nevts_after      = dst[mask]      .event.nunique()
+    nevts_before     = dst[input_mask].event.nunique()
+    eff              = nevts_after / nevts_before
+    all_in_range(data         = np.array(eff)  ,
+                 minval       = eff_interval[0],
+                 maxval       = eff_interval[1],
+                 display_name = column.value   ,
+                 strictness   = strictness     ,
+                 right_closed = True)
+
+    return mask
+
+
+def select_band_and_check(dst         : pd.DataFrame                                              ,
+                          boot_map    : ASectorMap                                                ,
+                          norm_strat  : Optional[NormStrategy]                  = NormStrategy.max,
+                          input_mask  : Optional[np.ndarray]                    = None            ,
+                          range_DT    : Optional[Tuple[np.ndarray, np.ndarray]] = (10, 1300)      ,
+                          range_E     : Optional[Tuple[np.ndarray, np.ndarray]] = (10.0e+3,14e+3) ,
+                          nsigma_sel  : Optional[float]                         = 3.5             ,
+                          eff_interval: Optional[Tuple[float, float]]           = [0,1]           ,
+                          strictness  : Optional[Strictness]                    = Strictness.raise_error
```

Comment:
Of these, I think `nsigma_sel` and `strictness` could/should be optional, the rest I prefer that they are not. I'm not sure about `norm_strat`, because there might be some assumptions in the code that are optimized for that choice. I think you will have a better feeling for that one.